public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -353,26 +356,39 @@ export class ElasticDatasource extends DataSourceApi<ElasticsearchQuery, Elastic
   }
 
   interpolateVariablesInQueries(queries: ElasticsearchQuery[], scopedVars: ScopedVars): ElasticsearchQuery[] {
-    let expandedQueries = queries;
-    if (queries && queries.length > 0) {
-      expandedQueries = queries.map((query) => {
-        const expandedQuery = {
-          ...query,
-          datasource: this.name,
-          query: this.interpolateLuceneQuery(query.query || '', scopedVars),
+    // We need a separate interpolation format for lucene queries, therefore we first interpolate any
+    // lucene query string and then everything else
+    const interpolateBucketAgg = (bucketAgg: BucketAggregation): BucketAggregation => {
+      if (bucketAgg.type === 'filters') {
+        return {
+          ...bucketAgg,
+          settings: {
+            ...bucketAgg.settings,
+            filters: bucketAgg.settings?.filters?.map((filter) => ({
+              ...filter,
+              query: this.interpolateLuceneQuery(filter.query || '', scopedVars),
+            })),
+          },
         };
+      }
 
-        for (let bucketAgg of query.bucketAggs || []) {
-          if (bucketAgg.type === 'filters') {
-            for (let filter of bucketAgg.settings?.filters || []) {
-              filter.query = this.interpolateLuceneQuery(filter.query, scopedVars);
-            }
-          }
-        }
-        return expandedQuery;
-      });
-    }
-    return expandedQueries;
+      return bucketAgg;
+    };
+
+    const expandedQueries = queries.map(
+      (query): ElasticsearchQuery => ({
+        ...query,
+        datasource: this.name,
+        query: this.interpolateLuceneQuery(query.query || '', scopedVars),
+        bucketAggs: query.bucketAggs?.map(interpolateBucketAgg),
+      })
+    );
+
+    const finalQueries: ElasticsearchQuery[] = JSON.parse(
```

Comment:
Why do we need `interpolateBucketAgg`? Wouldn't `this.templateSrv.replace(JSON.stringify(expandedQueries), scopedVars)` interpolate it correctly?

Reply:
because if it a bucketAgg is `filters` we need to use `interpolateLuceneQuery` in each `settings.filters[n].query` property. 

`interpolateBucketAgg` is a noop for all other aggregations

Reply:
Oh I got it now. It's because we need to default to `*`. It's actually covered in tests but it's kind of tricky to see it straight away. IMO it'd be a bit easier if it was more an integration test using the real `TemplateSrv` and an extra case to explicitly pass vars that have empty values - something like this:
https://github.com/grafana/grafana/commit/229d0c8a82f9f6dafe224c03e496ba7be70f4a74 
But yeah, the coverage should be the same anyway so it's up to you 👍

Reply:
actually `interpolateLuceneQuery` uses a different interpolation format (https://grafana.com/docs/grafana/latest/variables/advanced-variable-format-options/#lucene---elasticsearch) AND default to `*` if the interpolation results in empty strings.

> and an extra case to explicitly pass vars that have empty values

you mean something like: 
https://github.com/grafana/grafana/blob/bad048b7ba1fecd28e76509c545a67fc8b70fdeb/public/app/plugins/datasource/elasticsearch/datasource.test.ts#L880

?

Reply:
> vars that have empty values

aaaaaah, I got your point now, lemme check. 👍

Reply:
i took a better look at it and it should be already covered, I don't feel it will ad much value adding it here, so I'm gonna merge it :)